spyder/plugins/ipythonconsole/tests/test_ipythonconsole.py

```diff
@@ -2516,5 +2516,37 @@ def test_filter_frames_in_tracebacks(ipyconsole, qtbot, tmp_path):
     assert "BdbQuit" not in control.toPlainText()
 
 
+def test_lowercase_wdir(ipyconsole, qtbot, tmp_path):
+
+    shell = ipyconsole.get_current_shellwidget()
+    control = shell._control
+    qtbot.waitUntil(
+        lambda: shell._prompt_html is not None, timeout=SHELL_TIMEOUT
+    )
+    dir_a = tmp_path / 'SensitiveCaseA'
+    dir_a.mkdir()
+    filename_a = dir_a / 'a.py'
+    filename_a.write_text('a = 1')
+
+    # Filename in the format used when running magics from the main toolbar
+    fname = str(filename_a).replace('\\', '/')
+    fdir = str(dir_a).replace('\\', '/')
+
+    # Run file
+    with qtbot.waitSignal(shell.executed):
+        shell.execute(f"%runfile {fname} --wdir {fdir}")
+
+    with qtbot.waitSignal(shell.executed):
+        shell.execute("%clear")
+
```

Comment:
```suggestion

    empty_console_text = '\n\nIn [3]: ' if os.name == "nt" else '\nIn [3]: '
    qtbot.waitUntil(lambda: empty_console_text == shell._control.toPlainText())
```

Wait until the console is really empty.